MembraneVideoroomDemo/Controllers/RoomController.swift

```diff
@@ -142,6 +146,15 @@ class RoomController: ObservableObject {
                 room.updateTrackMetadata(
                     trackId: trackId, trackMetadata: .init(["active": isMicEnabled, "type": "audio"]))
             }
+            guard var p = participants[localParticipantId] else {
+                return
+            }
+            let pv = findParticipantVideoByOwner(participantId: localParticipantId)
+
+            p.isAudioTrackActive = isMicEnabled
+            participants[localParticipantId] = p
+            pv?.participant = p
+            self.objectWillChange.send()
```

Comment:
I'm not a fan of calling `self.objectWillChange.send()` every time something changes. It may be redundant or you may forget about calling it, leading to bugs. Instead, changing Published variables should update what's necessary automatically.